### ISSUE TITLE
proposal: use TTL and last access time to evict unused versions from service-worker's cache

### ISSUE THREAD

**Issue by slavafomin** [angular/angular#41508]
# 🚀 feature request

Hello!

### Relevant Package
This feature request is for `@angular/service-worker`.

### Description
Right now, the [cleanupCaches()](https://github.com/angular/angular/blob/3a823abcc554cffdda1e14e181dadad9063701de/packages/service-worker/worker/src/driver.ts#L890) function from the service-worker implementation evicts versions of the application that are not present in the [Clients](https://developer.mozilla.org/en-US/docs/Web/API/Clients) list.

However, as stated in the comment for this function:

> This may skip some clients in the browser back-forward cache, but not much can be done about that.

### Describe the solution you'd like
I thought, that we could use a reasonable TTL to make the cache eviction a little bit safer. Namely:

1. we assign a `last access time` property to each version of the application and update it each time when the specific version is used by the client.
2. we configure a reasonable TTL value, e.g. 24 hours
3. during the cache cleanup we add an additional step to determine if the version could be safely evicted from the cache by comparing `last access time` for this version with the TTL.

In other words: the application could be removed from the cache when the assigned client ID is no longer present and after the specific time has passed since the last use of this version.

This will ensure that even if a browser has cached some version of the application in it's back-forward cache it can still be used (during the specified TTL period).

I'm willing to implement this feature if it will be considered useful.

### Describe alternatives you've considered
Don't know about any alternative solutions.

**Comment by gkalpak** [comment]
That doesn't sound unreasonable. Although I would prefer to make it opt-in or opt-out with a smaller thershold. So, I think it would be better to make this configurable (per version).

If you want to work on a PR, @slavafomin, I would be happy to review :wink:
